nevergrad/optimization/optimizerlib.py

```diff
@@ -1425,33 +1425,105 @@ def _internal_tell_not_asked(self, candidate: p.Parameter, value: float) -> None
+        if parent is self.parametrization:
+            candidate.heritage["lineage"] = candidate.uid
+        candidate._meta["sigma"] = self.sigma
+        return candidate
 
     def _internal_tell_candidate(self, candidate: p.Parameter, value: float) -> None:
         candidate._meta["loss"] = value
-        self.popsize.add_value(value)
         self.children.append(candidate)
-        if len(self.children) >= self.popsize.llambda:
+        if len(self.children) >= self.llambda:
             # Sorting the population.
             self.children.sort(key=lambda c: c._meta["loss"])
             # Computing the new parent.
-            mu = self.popsize.mu
-            self.current_center = sum(c.get_standardized_data(reference=self.parametrization)  # type: ignore
-                                      for c in self.children[: mu]) / mu
-            t1 = [(c.get_standardized_data(reference=self.parametrization) - self.current_center)**2 for c in self.children[: mu]]
+            self.parents = self.children[: self.mu]
+            self.children = []
+            self.current_center = sum(c.get_standardized_data(reference=self.parametrization) for c in self.parents) / self.mu  # type: ignore
             # EMNA update
-            self.sigma = np.sqrt(sum(t1) / (mu))
-            imp = max(1, (np.log(self.popsize.llambda) / 2)**(1 / self.dimension))
-            if self.num_workers / self.dimension > 16:
+            stdd = [(self.parents[i].get_standardized_data(reference=self.parametrization) - self.current_center)**2 for i in range(self.mu)]
+            if self.isotropic:
+                self.sigma = np.sqrt(sum(stdd) / (self.mu * self.dimension))
+            else:
+                self.sigma = np.sqrt(np.sum(stdd, axis=1) / (self.mu))
+
+            if self.num_workers / self.dimension > 16: # faster decrease of sigma if large parallel context
+                imp = max(1, (np.log(self.llambda) / 2)**(1 / self.dimension))
                 self.sigma /= imp
-            self.parents = self.children[: mu]
-            self.children = []
 
+    def _internal_tell_not_asked(self, candidate: p.Parameter, value: float) -> None:
+        base.TellNotAskedNotSupportedError
+
+
+class EMNA(base.ConfiguredOptimizer):
+    """Estimation of Multivariate Normal Algorithm
+        This algorithm is quite efficient in a parallel context, i.e. when
+        the population size is large.
+
+        Parameters
+        ----------
+        - `"isotropic"`: if true: isotropic version on EMNA, i.e. we have an
+                                    identity matrix for the Gaussian
+                         else: anisotropic version, we here consider the
+                                    separable version, meaning we have a diagonal
+                                    matrix for the Gaussian
+        - `"naive"`: optimistic version. This parameter can be set to true for
+                    noiseless problems
```

Comment:
A few details for the documentation to be able to be correctly generated:
- first the parameters should be presented slightly differently
```suggestion
        isotropic: bool
             isotropic version on EMNA if True, i.e. we have an
             identity matrix for the Gaussian, else  we here consider the separable
             version, meaning we have a diagonal matrix for the Gaussian (anisotropic)
        naive: bool
             set to False for noisy problem, so that the best points will be an average of the final
             population.
```
- then notice (if that makes sense), I changed the way you describe "naive", since by default it is True
- finally, thee whole docstring (except the first line, is overindented: all lines should be aligned with the initial `"""`

Reply:
Ok thx ! I think it is ok now.
Just the test deactivation remains, but I am not sure on how to correct it. It seems for me that the test will fail for all variants. I have just tried to put algorithms in the "SLOW" list into this test and they fail too, so I don't know. From here, as I am not sure to understand why I would be tempted to add EMNA in the SLOW list, in that way it will skip the test, but I think you will disagree :-)